docs/developer-guide/aggregate-streaming-data.rst

```diff
@@ -0,0 +1,110 @@
+.. _aggregate-streaming-data-with-ksql:
+
+Aggregate Streaming Data With KSQL
+##################################
+
+KSQL supports several aggregate functions, like COUNT and SUM. You can use
+these to build stateful aggregates on streaming data. For the full list, see
+:ref:`ksql_aggregate_functions`.
+
+Here are some examples that show how to aggregate data from an inbound stream
+of pageview records, named ``pageviews``.
+
+Aggregate Results in a KSQL Table
+*********************************
+
+The result of an aggregate query in KSQL is always a table, because KSQL
+computes the aggregate for each key, and possibly for each window per key, and
+updates these results as it processes new input data.

```

Comment:
Nit: "updates these results as it processes new input data" -> "updates these results as it processes new input data for a key".

To re-emphasize the point that, in KSQL, aggregates are always computed per key. (You already alluded to this prior in the paragraph.)